src/libnnstreamer-edge/nnstreamer-edge-common.h

```diff
@@ -45,6 +45,28 @@ extern "C" {
 #define nns_edge_lock(h) do { pthread_mutex_lock (&(h)->lock); } while (0)
 #define nns_edge_unlock(h) do { pthread_mutex_unlock (&(h)->lock); } while (0)
 
+/**
+ * @brief Internal data structure for metadata.
+ */
+typedef struct _nns_edge_metadata_node_s nns_edge_metadata_node_s;
+
+/**
+ * @brief Internal data structure for metadata.
+ */
+struct _nns_edge_metadata_node_s {
+  char *key;
+  char *value;
+  nns_edge_metadata_node_s *next;
+};
+
+/**
+ * @brief Internal data structure to handle metadata. This struct should be managed in the handle.
+ */
+typedef struct {
+  unsigned int list_len;
+  nns_edge_metadata_node_s *list;
+} nns_edge_metadata_s;
```

Comment:
Need a lock to protect if the functions of nnstreamer-edge-common can be accessed from different threads.

Reply:
I agree with myungjoo's opinion.  The result of `nns_edge_metadata_find()` would vary if `nns_edge_metadata_set()` function is running in another thread. I thinks that the read lock mechanism is good enough for this issue.

Reply:
Suggestion: adding lock in edge-data handle.
- metadata is internal struct.
- metadata is internally used in edge handle and edge-data handle, then called with edge handle lock. However edge-data handle does not have lock now.

Reply:
That's be ok.

Reply:
I agreed with jaeyun-jung  's opinion. 👍